backend/hubs/models/hub.py

```diff
@@ -276,6 +285,42 @@ class Hub(models.Model):
         blank=True,
     )
 
+    parent_hub = models.ForeignKey(
+        "self",
+        help_text="Points to the parent hub if this is a sub-hub",
+        verbose_name="Parent Hub",
+        related_name="sub_hubs",
+        blank=True,
+        null=True,
+        on_delete=models.SET_NULL,
+    )
+
+    # Constraints to ensure that the hub is a parent or a sub-hub
+    def clean(self):
+        if self.parent_hub:
+            # ensure no self loop
+            if self == self.parent_hub:
+                raise ValidationError("A hub cannot be its own parent.")
+
+            # hub is now a child. Ensure that it is not a parent hub
+            children = Hub.objects.filter(parent_hub=self)
+            if children.exists():
+                raise ValidationError(
+                    "This hub cannot be a parent hub because it is already a sub-hub of another hub. child: {}| parent: {}".format(
+                        self.parent_hub.name, children.first().name
+                    )
+                )
+
+            # ensure that the parent hub is not a child of another hub
+            if self.parent_hub.parent_hub:
+                raise ValidationError(
+                    "A hub cannot be a parent-hub of another parent-hub. Please ensure that the parent hub is a top-level hub. self: {}| parent: {}| parents parent: {}".format(
+                        self.name, self.parent_hub.name, self.parent_hub.parent_hub.name
+                    )
+                )
+
+        return super().clean()
```

Comment:
overwriting the inherited `BaseModel.clean()` function that validates whether a object creation or update is valid. 
It will be run with the new changes applied and shall throw in case of an error state.

1. check if the current model itself is valid (no loops, not child and parent at the same time) 
2. check if the new parent is valid (parent does not have another parent)

Ensuring this via the database layer was more complicated via Django ORM than this.